core_test.go

```diff
@@ -876,6 +884,55 @@ func TestMatMax(t *testing.T) {
 	}
 }
 
+// This test does the following:
+// - download the JPEG image
+// - decodes to the image.Image with magic number
+// - writes its content (bytes) to the temp file
+// - creates Mat from go image
+// - reads new Mat from temp file path
+// - runs diff
+func TestMatFromImage(t *testing.T) {
+	url := "http://i.imgur.com/m1UIjW1.jpg"
+	response, err := http.Get(url)
```

Comment:
I would avoid a test case that relies on network. It is not reproducible. Can you reuse an existing image maybe (part of this repo, see: https://github.com/hybridgroup/gocv/tree/master/images)?